diff --git a/libmamba/CMakeLists.txt b/libmamba/CMakeLists.txt
index 366aa90f..2729db4d 100644
--- a/libmamba/CMakeLists.txt
+++ b/libmamba/CMakeLists.txt
@@ -276,12 +276,6 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
                 libcurl.a
                 libssh2.a
 
-                libgssapi_krb5.a
-                libkrb5.a
-                libk5crypto.a
-                libkrb5support.a
-                libcom_err.a
-
                 libssl.a
                 libcrypto.a
 
@@ -289,17 +283,12 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
 
                 libiconv.a
                 libbz2.a
-                liblz4.a
                 libzstd.a
                 libz.a
-                liblzma.a
-                liblzo2.a
 
                 libsolv.a
                 libsolvext.a
 
-                libnghttp2.a
-
                 libyaml-cpp.a
 
                 libreproc++.a
@@ -309,6 +298,9 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
                 set(REQUIRED_STATIC_DEPS ${REQUIRED_STATIC_DEPS} libc++.a)
             endif()
 
+            if (UNIX AND NOT APPLE)
+                list(REMOVE_ITEM REQUIRED_STATIC_DEPS libiconv.a)
+            endif()
 
             set(STATIC_DEPS "")
             foreach(LIB ${REQUIRED_STATIC_DEPS})
@@ -321,9 +313,10 @@ macro(libmamba_create_target target_name linkage deps_linkage output_name)
 
             if (APPLE)
                 find_library(SECURITY_LIBRARY Security)
+                find_library(SYSTEMCONFIGURATION_LIBRARY SystemConfiguration)
                 find_library(COREFOUNDATION_LIBRARY CoreFoundation)
                 message("Found library: ${SECURITY_LIBRARY}\n${COREFOUNDATION_LIBRARY}")
-                list(APPEND STATIC_DEPS ${COREFOUNDATION_LIBRARY} ${SECURITY_LIBRARY})
+                list(APPEND STATIC_DEPS ${COREFOUNDATION_LIBRARY} ${SECURITY_LIBRARY} ${SYSTEMCONFIGURATION_LIBRARY})
             endif()
 
             message("   -> Found static dependencies:")
